docs/git/getting-started.md

```diff
@@ -0,0 +1,176 @@
+
+If you make a commit and then run:
+```
+git log --format=raw
+```
+it should be obvious if your commits are GPG signed.
+
+## Branching from develop
+
+Cardano development happens on the "develop" branch. You can find out what
+branch you are one by running:
+```
+git branch
+```
+which will list all the branches in your local repository and print an asterisk
+to the left of the current branch. You can list all the branches in the
+Github repository using:
+```
+git branch --all
+```
+
+Once you are ready to start working on a new feature you should create a branch
+to work on that feature. To name space things correctly, its best to name the
+branch with a combination of your username and the YouTrack ticket name. Assuming
+you are on the "develop" branch, to create the new branch and then change to
+it:
+```
+git branch username/YT-1234
+git checkout username/YT-1234
+```
+You can then work on your changes and commit them using something like:
+```
+git commit --message="YT-1234: Fixed the wibble"
+```
+which includes the YouTrack ticket number in the commit message. Its a good
+idea not to put too much information in the title of the git commit message.
+If you want to add more information you can run:
+```
+git commit --amend
+```
+which will bring up you text editor allowing you to add a message body to
+the text after the commit message title like for instance
+[this commit](https://github.com/input-output-hk/cardano-sl/commit/ddc58940e5ddec357cc94097df2951359cb37ed5)
+
+Once you have committed your change you can use:
+```
+git show
+```
+to show the contents of the commit. Similarly, you can use something like
+```
+git show ddc58940e5
+```
+to show the contents of the commit whose hash begins with "ddc58940e5"
+
+
+## Customizing Git
+
+Git has a configuration file at `${HOME}/.gitconf` which might look like this:
+```
+[user]
+  name = My Name
+  email = me@example.com
+
+[core]
+  whitespace = trailing-space,space-before-tab
+  autocrlf = false
+
+[github]
+  user = gh-user-name
+
+[alias]
+  # Simple aliases
+  br = branch
+  co = checkout
+  cp = cherry-pick
+  st = status
+
+  # More complex aliases
+  logv = log --name-status
+  logp = "log --graph --pretty=format:\"%h -%d %s (%cr)\" --abbrev-commit --date=relative"
+  subup = submodule update
+
+  # Check create a branch off the current branch and then change to it.
+  # eg: "git bct my-name/my-topic"
+  bct = "checkout -b"
+
+[pull]
+  # Make "git pull --rebase" the default behaviour for "git pull"
+  rebase = true
+
+[push]
+  default = matching
+
+[diff]
+  ignoreSubmodules = dirty
+  compactionHeuristic = true
+
+[pager]
+  status = false
+  branch = false
+  stash = false
+  diff = true
+```
+
+Its also possible to define shell functions that string together one of more
+git commands. For example, the following shell function when added to the users
+`${HOME}/.bashrc` file allows rebase the current branch against the develop
+branch on Github:
```

Comment:
There's a pretty good list of aliases for git here: https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/git/git.plugin.zsh#L41

I also have these ones:

```
alias glog="git log --pretty=format:'%Cred%h% %Creset - %Cgreen%<(12)%cr%Creset %<(60,trunc)%s %Cblue%an%Creset %C(yellow)%d%Creset'"
alias glogg="git log --graph --pretty=format:'%Cred%h% %Creset - %Cgreen%<(12)%cr%Creset %<(60,trunc)%s %Cblue%an%Creset %C(yellow)%d%Creset'"
```

Reply:
In a similar fashion:

```
[alias]
  lg = log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit
```

gets you a pretty neat graph visualization of the commit history:

![image](https://user-images.githubusercontent.com/5680256/44022024-b94e60f0-9ee6-11e8-8c10-72a44fd3d106.png)